src/registry/parseRegistry.ts

```diff
@@ -1,21 +1,59 @@
-// Copyright 2023 Parity Technologies (UK) Ltd.
+// Copyright 2023-2024 Parity Technologies (UK) Ltd.
 
 import { ASSET_HUB_CHAIN_ID } from '../consts';
-import type { AssetTransferApiOpts } from '../types';
-import type { ChainInfoRegistry } from './types';
+import type { AssetTransferApiInjectedOpts } from '../types';
+import type { ChainInfo, ChainInfoRegistry, InjectedChainInfo } from './types';
 
-export const parseRegistry = (registry: ChainInfoRegistry, assetsOpts: AssetTransferApiOpts): ChainInfoRegistry => {
+const updateRegistry = (injectedChain: InjectedChainInfo, registry: ChainInfoRegistry, registryChain: string) => {
```

Comment:
For a cleaner solution we should really use recursion here for deep comparisons and updates. Having a triple nested loop is not ideal.

Reply:
That being said there are arguments to both sides:

recursion: We could implement it once in a generic way, and never have to touch it again, even if the structure of the registry changes.

iteration: It's generally faster, and less memory intensive especially in stack frames. That being said we would have to update it every time there is a change in the registry.

So I am definitely up for discussion on this.